### ISSUE TITLE
AssertionError Need to set transaction when entering span!

### ISSUE THREAD

**Issue by aveeday** [getsentry/sentry-python#472]
After upgrading from sentry-sdk 0.10.2 to sentry-sdk-0.11.0 I received many errors with next text:
```
AssertionError
Need to set transaction when entering span!
```
Sometimes some requests leads to that error, but after repeating the same ones they work without error

asgiref                  3.1.2
channels                 2.2.0     
channels-redis           2.4.0
Django                   2.2.4
djangorestframework      3.9.1
psycopg2-binary          2.8.3
redis                    2.10.6

**Comment by untitaker** [comment]
Hi, are you using the ASGI Middleware of the Sentry sdk?

**Comment by untitaker** [comment]
In any case I converted that assertion into a warning, but I'd like to figure out this issue anyway.

**Comment by aveeday** [comment]
I do not use ASGI Middleware.

Tracebacks:
```
AssertionError: Need to set transaction when entering span!
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "rest_framework/generics.py", line 241, in get
    return self.list(request, *args, **kwargs)
  File "rest_framework/mixins.py", line 42, in list
    page = self.paginate_queryset(queryset)
  File "rest_framework/generics.py", line 173, in paginate_queryset
    return self.paginator.paginate_queryset(queryset, self.request, view=self)
  File "rest_framework/pagination.py", line 325, in paginate_queryset
    self.count = self.get_count(queryset)
  File "rest_framework/pagination.py", line 466, in get_count
    return queryset.count()
  File "django/db/models/query.py", line 392, in count
    return self.query.get_count(using=self.db)
  File "django/db/models/sql/query.py", line 504, in get_count
    number = obj.get_aggregation(using, ['__count'])['__count']
  File "django/db/models/sql/query.py", line 489, in get_aggregation
    result = compiler.execute_sql(SINGLE)
  File "django/db/models/sql/compiler.py", line 1100, in execute_sql
    cursor.execute(sql, params)
  File "contextlib.py", line 119, in __exit__
    next(self.gen)
```

```
AssertionError: Need to set transaction when entering span!
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 483, in dispatch
    self.initial(request, *args, **kwargs)
  File "rest_framework/views.py", line 400, in initial
    self.perform_authentication(request)
  File "rest_framework/views.py", line 326, in perform_authentication
    request.user
  File "rest_framework/request.py", line 223, in user
    self._authenticate()
  File "rest_framework/request.py", line 376, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
  File "rest_framework/authentication.py", line 192, in authenticate
    return self.authenticate_credentials(token)
  File "rest_framework/authentication.py", line 197, in authenticate_credentials
    token = model.objects.select_related('user').get(key=key)
  File "django/db/models/query.py", line 402, in get
    num = len(clone)
  File "django/db/models/query.py", line 256, in __len__
    self._fetch_all()
  File "django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "django/db/models/sql/compiler.py", line 1100, in execute_sql
    cursor.execute(sql, params)
  File "contextlib.py", line 119, in __exit__
    next(self.gen)
```

**Comment by untitaker** [comment]
@aveeday your issue should be "fixed" with 0.11.1.
